src/adiabatic/julip.jl

```diff
@@ -9,8 +9,9 @@ export JuLIPModel
 
 Model for interfacing with JuLIP potentials.
 """
-struct JuLIPModel{T} <: AdiabaticModel
+struct JuLIPModel{T,M} <: AdiabaticModel
     atoms::JuLIP.Atoms{T}
+    constr::Matrix{M}
```

Comment:
Probably it is fine here to require that the type parameter T of the atoms is the same as that for the matrix. Or if the constraints matrix is supposed to always contain ones and zeros I would suggest using `Matrix{Int}` instead.